ui/app/hooks/useMethodData.js

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useCallback } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { getContractMethodData as getContractMethodDataAction } from '../store/actions'
+
+import { getKnownMethodData } from '../selectors/selectors'
+
+/**
+ * useMethodData
```

Comment:
Nit: can we add a single line description of the function here instead? Maybe extracted from the larger description below.

Reply:
Do you like: 
``` * Access known method data and attempt to resolve unknown method data ``` for this?